### DISCUSSION TITLE
Connect two Pico boards through USB hub

### DISCUSSION THREAD

**Discussion by mars-low** [fhdm-dev/scoppy#36] [General]
I've planned to use Scoppy together with [Serial USB Terminal](https://play.google.com/store/apps/details?id=de.kai_morich.serial_usb_terminal) . I own two Pico boards and often connect both at the same time through USB hub. Previously I've been running CircuitPython on each Pico and interacted with them through serial USB terminal. I've been able to control both in separate app windows on splitted screen. For example I was able to play with PWM signal on first Pico and read ADC values on second one. Now I would like to replace firmware on one Pico with [scoppy-pico](https://github.com/fhdm-dev/scoppy-pico) due to wonderful app interface and keep CircuitPython on second one for experiments with available API through Serial USB Terminal app mentioned above. 

At the moment I am unable to do that, because Scoppy app keeps asking about permission to access Pico, but actually doesn't connect at all. Similarly, if I am connected successfully and then I attach second Pico to USB hub, Scoppy app will loose connection. I suspect that Scoppy app looks for compatible device based on USB PID and VID values and when it sees two available Pico boards it can't recognize which board actually runs scoppy-pico firmware. 
I am writing just out of curiosity, because I think it may be Android app feature to ignore second Pico when first is connected successfully. In the meantime as a workaround I'm using some old tablet to run Serial USB Terminal and phone with Android 9 to run Scoppy oscilloscope.

**Comment by fhdm-dev** [comment]
Hi. 

Yes. You are correct. The app doesn't know which of the boards to connect to. It just assumes any connected USB device (that is a CDC serial device) is running the scoppy-pico firmware. Oops! 

I'll see what I can do to get it working. Hopefully in the next app update :smile:
